### PR TITLE
dm: update doc for optimistic shard ddl

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

as title

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/9317
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/feature-shard-merge-optimistic.md [hunk 1]
```diff
@@ -35,18 +35,28 @@ It takes some risks to use the optimistic mode. Follow these rules when you use
...
 
-Currently, the following statements are not supported in the optimistic mode:
+In the optimistic mode, most of the DDL statements executed in the upstream are automatically migrated to the downstream with no extra effort required. These DDL statements are called "Type 1 DDL".
 
-- `ALTER TABLE table_name ADD COLUMN column_name datatype NOT NULL` (To add a `NOT NULL` column without a default value).
-- `ALTER TABLE table_name ADD COLUMN column_name datetime DEFAULT NOW()` (To add a column with a varying value).
-- `ALTER TABLE table_name ADD COLUMN col1 INT, DROP COLUMN col2` (Contains both `ADD COLUMN` and `DROP COLUMN` in one DDL statement).
-- `ALTER TABLE table_name RENAME COLUMN column_1 TO column_2;` (To rename a column).
-- `ALTER TABLE table_name RENAME INDEX index_1 TO index_2;` (To rename an index).
+DDL statements that change the column name, the column type, or the column default value are called "Type 2 DDL". When you execute Type 2 DDL statements in the upstream, make sure that the DDL statements are executed in all sharded tables in the same order.
```
Comment (hfxsd) [inline dm/feature-shard-merge-optimistic.md:40]:
```suggestion
DDL statements that change the column name, the column type, or the column default value are called "Type 2 DDL". When you execute Type 2 DDL statements in the upstream, make sure that you execute the DDL statements in all sharded tables in the same order.
```

File: dm/feature-shard-merge-optimistic.md [hunk 2]
```diff
@@ -35,18 +35,28 @@ It takes some risks to use the optimistic mode. Follow these rules when you use
```
Comment (hfxsd) [inline dm/feature-shard-merge-optimistic.md:35]: 上面 L34 有个typo：You do not need to executed

**Review by hfxsd — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**